python/hidet/option.py

```diff
@@ -46,9 +48,24 @@ def register_option(
     checker: Optional[Callable[[Any], bool]] = None,
 ):
     registered_options = OptionRegistry.registered_options
+    doc = OptionRegistry.config_doc
     if name in registered_options:
         raise KeyError(f'Option {name} has already been registered.')
     registered_options[name] = OptionRegistry(name, type_hint, description, default_value, normalizer, choices, checker)
+    doc.add(tomlkit.comment(description))
+    if choices is not None:
+        doc.add(tomlkit.comment(f'  choices: {choices}'))
+    if isinstance(default_value, (bool, int, float, str)):
+        doc.add(name, default_value)
+    elif isinstance(default_value, Tuple):
+        # represent tuples are toml arrays, do not allow python lists are default values to avoid ambiguity
+        val = list(default_value)
+        arr = tomlkit.array()
+        arr.extend(val)
+        doc.add(name, arr)
+    else:
+        raise ValueError(f'Invalid type of default value for option {name}: {type(default_value)}')
+    doc.add(tomlkit.nl())
```

Comment:
Consider moving this logic into a standalone function like `_create_toml_doc()`.